src/api/telemetry/TelemetryAPI.js

```diff
@@ -324,6 +326,17 @@ define([
             return Promise.reject('No provider found');
         }
 
+        const hasRequestProvider = Object.hasOwn(provider, 'request');
```

Comment:
I don't think the logic is quite right here. This will check whether the historical provider that matched the given domain object has a request function defined, and then caches the result for all domain objects.

Instead I think what we want to do is check whether there are _any_ historical providers defined at all (for _any_ domain objects). The trick here is that `supportsRequest` accepts a domain object, so how do we check for all domain objects? I think you're on the right track with testing whether `request` is defined. We should just do this check for all registered telemetry providers.

We should do this check before the check for the _specific_ domain object, and then cache the result. If no historical providers are defined at all, then we should resolve with an empty array.

Reply:
is it per `domainObject` or per `object type`? I think we can only define request per plugin/type and not per domainObject.

Reply:
* Checking for the presence of a `request` function for a given provider is redundant. This is the role of `supportsRequest`. If `this.findRequestProvider()` has returned a provider, then it supports request. So this check isn't necessary.
* Right now we reject the promise if there is no historical provider for a given object, on line 320. Instead we should resolve the promise with an empty array and print a warning to the console.
* There is a special use case where a developer wants to use Open MCT in real-time mode only. eg. they don't have a telemetry archive at all. In this case we don't want to flood the console with warnings. Instead we can shortcut the whole function and always just resolve with an empty array. How do we know if this is the case? The best way I can think of is to loop through **all** of the telemetry providers and check to see if there are **any** telemetry providers that define a `request` function at all (ie for any object). We can cache the result of this so we don't need to run the check each time.

Reply:
fixed promise issue in `platform/telemetry/src/TelemetryCapability.js`
Currently its always throw error if `telemetryservice` is `undefined`, @akhenry should we cache it per DomainObject and show only once? as mentioned in previous solution?